backend/model/model.go

```diff
@@ -2109,6 +2110,21 @@ type LogAlertEvent struct {
 	SentAt     time.Time
 }
 
+type SavedSegmentEntityType string
+
+const (
+	SavedSegmentEntityTypeLog   SavedSegmentEntityType = "Log"
+	SavedSegmentEntityTypeTrace SavedSegmentEntityType = "Trace"
+)
+
+type SavedSegment struct {
+	Model
+	Name       string
+	EntityType SavedSegmentEntityType `gorm:"index:idx_saved_segment"`
+	Params     string                 `json:"params"`
+	ProjectID  int                    `gorm:"index:idx_saved_segment" json:"project_id"`
```

Comment:
Add an index here since most queries will be by `project_id` and `entity_type` (e.g. "Log saved segments for project 1", "Trace saved segments for Project 23")